stdlib/buffer.gr

```diff
@@ -96,674 +96,691 @@ let checkIsIndexInBounds = (i, len, buf) => {
+ * @param bytes: The byte sequence to append
  * @param buffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let addFloat32 = (value, buffer) => {
-  autogrow(_32BIT_LEN, buffer)
-  let index = buffer.len
-  buffer.len = buffer.len + _32BIT_LEN
-  setFloat32(index, value, buffer)
+@disableGC
+export let rec addBytesSlice = (start: Number, length: Number, bytes: Bytes, buffer: Buffer) => {
+  Memory.incRef(WasmI32.fromGrain(autogrow))
+  Memory.incRef(WasmI32.fromGrain(length))
+  Memory.incRef(WasmI32.fromGrain(buffer))
+  autogrow(length, buffer)
+
+  let srcOff = coerceNumberToWasmI32(start)
+  let dstOff = coerceNumberToWasmI32(buffer.len)
+  let src = WasmI32.fromGrain(bytes)
+  let dst = WasmI32.fromGrain(buffer.data)
+  appendBytes(srcOff, dstOff, coerceNumberToWasmI32(length), src, dst)
+
+  Memory.incRef(WasmI32.fromGrain((+)))
+  Memory.incRef(WasmI32.fromGrain(buffer.len))
+  Memory.incRef(WasmI32.fromGrain(length))
+  buffer.len = buffer.len + length
+
+  Memory.decRef(WasmI32.fromGrain(start))
+  Memory.decRef(WasmI32.fromGrain(length))
+  Memory.decRef(WasmI32.fromGrain(bytes))
+  Memory.decRef(WasmI32.fromGrain(buffer))
+  Memory.decRef(WasmI32.fromGrain(addBytesSlice))
+  void
 }
 
 /**
- * Gets a signed 64-bit integer starting at the given byte index.
+ * Appends the bytes of a source buffer to destination buffer.
  *
- * @param index: The byte index to access
- * @param buffer: The buffer to access
- * @returns A signed 64-bit integer that starts at the given index
+ * The source buffer is not mutated by this operation. The destination buffer, however, is mutated.
+ *
+ * @param srcBuffer: The buffer to append
+ * @param dstBuffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let getInt64 = (index, buffer) => {
-  checkIsIndexInBounds(index, _64BIT_LEN, buffer)
-  Bytes.getInt64(index, buffer.data)
+export let addBuffer = (srcBuffer, dstBuffer) => {
+  addBytesSlice(0, srcBuffer.len, srcBuffer.data, dstBuffer)
 }
 
 /**
- * Sets a signed 64-bit integer starting at the given byte index.
+ * Appends the bytes of a part of a buffer to the end of the buffer
  *
- * @param index: The byte index to update
- * @param value: The value to set
- * @param buffer: The buffer to mutate
+ * The source buffer is not mutated by this operation. The destination buffer, however, is mutated.
+ *
+ * @param start: The byte offset into the buffer
+ * @param length: The number of bytes to append
+ * @param srcBuffer: The buffer to append
+ * @param dstBuffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let setInt64 = (index, value, buffer) => {
-  checkIsIndexInBounds(index, _64BIT_LEN, buffer)
-  Bytes.setInt64(index, value, buffer.data)
+export let addBufferSlice = (start, length, srcBuffer, dstBuffer) => {
+  addBytesSlice(start, length, srcBuffer.data, dstBuffer)
 }
 
 /**
- * Appends a signed 64-bit integer to a buffer.
+ * @section Raw bytes: Functions for working with the raw bytes in a buffer.
```

Comment:
@jozanza @ospencer this `@section` probably needs a better title and description. Thoughts?

Reply:
OCaml calls this section `Binary encoding/decoding of integers`. I think that's pretty fair!

Reply:
```suggestion
 * @section Binary operations on integers: Functions for encoding and decoding integers stored in a buffer.
```
@ospencer how about this for title and description?